src/Halogen/Query/HalogenM.purs

```diff
@@ -163,8 +163,16 @@ getRef p = HalogenM $ liftF $ GetRef p identity
 raise :: forall s act ps o m. o -> HalogenM' s act ps o m Unit
 raise o = HalogenM $ liftF $ Raise o unit
 
-fork :: forall s act ps o m a. MonadAff m => HalogenM' s act ps o m a -> HalogenM' s act ps o m (Error -> m Unit)
-fork a = map liftAff <$> HalogenM (liftF $ Fork $ FF.fork a)
+newtype ForkId = ForkId Int
```

Comment:
Should constructor be exported?

Reply:
It's unavoidable, unfortunately - the driver creates/manages the IDs, so we can't make it private. It's the same thing with subscription IDs too.

Reply:
if only operation we care about the ID is to create initial id and increment it to calculate next one, then we might be able to hide constructor and export `initial = ForkId 0`  `next :: ForkId -> ForkId`

Reply:
Actually, I'm not sure that it's much better - a user can mess things up with access to a `succ` / `next` function just as much as they can be constructing an id manually 😕.

At least the re-exports for it from `Halogen` etc. only export that types and omit the constructors.

Reply:
Yes, that's also true  :(